### DISCUSSION TITLE
Warn when renovate does not lookup all the image tags

### DISCUSSION THREAD

**Discussion by carlossg** [renovatebot/renovate#31233] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, latest

### Please tell us more about your question or problem

When the docker datasource goes through 20 pages of tags it silently stops looking for more. This is confusing when renovate cannot find the new tag, at least it could show a warning

This is implemented in 
https://github.com/renovatebot/renovate/pull/31180

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
1. I don't agree that it should warn by default because in most cases it's harmless if tags are sorted by newest first. A debug log which you could remap to warn yourself would be better
2. #31180 was broken, and does not do what you intended to. You probably want to check for the presence of `url` after the loop completes
3. Remember that when results are cached and the lookup skipped, then you won't get a warning every time/every repo. You'd get a log message maybe once an hour each time the cache expired, and not for each repo.

**Reply by carlossg** [reply]
thanks, not concerned in the scheduled run but when running it to debug why a tag was not found I realized after having to debug through the code that we had way too many tags and the last page was reached

Would this be a good approach?
```
    if (!url && page >= pages) {
      logger.debug(
        { registryHost, dockerRepository },
        'Reached max pages for Docker tags lookup, may not have found all the image tags',
      );
    }
```

**Reply by rarkins** [reply]
The logic still seems wrong to me. I would have expected it to be simply `if (url) {`. i.e. if we finished paginating but there was still a next URL, it implies that there were more pages to fetch. The two reasons we can stop paginating are:
- We reached maximum pages (e.g. 20) so page === pages, or
- Nor more pages, i.e. `url` is empty

I think you care about the first one, where url should be non-empty

**Reply by carlossg** [reply]
sorry for the delay, I have tested it now and yes the correct code should be

```
        if (url) {
            logger.debug(
                { registryHost, dockerRepository },
                'Reached max pages for Docker tags lookup but there are more tags, may not have found all the image tags',
            );
        }
```
